src/main/java/org/primefaces/component/datatable/feature/ScrollFeature.java

```diff
@@ -82,7 +78,7 @@ public void encode(FacesContext context, DataTableRenderer renderer, DataTable t
 
     @Override
     public boolean shouldDecode(FacesContext context, DataTable table) {
-        return false;
+        return table.isScrollingRequest(context) && table.isSelectionEnabled();
```

Comment:
Why did you change all this features?

In the past we always decodde SelectionFeature, just if selection is enabled and indepdent if the current request is a selection request.
So in each request the SelectionFeature was called and decoded if the _selection param is available

Reply:
> Why did you change all this features?

I don't think I change anything, I just adapt according to my changes and move decode stuff in decode method instead of encode method. Arf now that I see it, loading data is also done in encode method... I'll have to change that

> In the past we always decodde SelectionFeature, just if selection is enabled and indepdent if the current request is a selection request.

Maybe, but as of today you have a few features just like this one that requires decoding selection (ex-findSelectedRowKeys). Right now, I just adapt according to my changes, tomorrow everything should probably be done in SelectionFeature

Reply:
Hmmm thats weird actually.
SelectionFeature#decode runs before all encode, so why does ScrollFeature needs a findSelectedRowKeys call?
Maybe we can remove this calls with your refactoring.

Reply:
I think those calls are to cover default selection, where items has been added right away to the `DataTable#selection` value. For example, you add a new row, and this row should be selected by default, that's what findSelectedRowKeys used to do (and I'm still doing) it's setting new rowKeys through `DataTable#selection` value. But I feel it would be a lot more simple if a new attribute was added to either DataTable or Column, to allow default selection

Reply:
That something for V11. I think its better to add this code to DataTableRenderer#preRender or similar isntead of in multiple XXXFeature which are not related to selection.

Reply:
Not that simple, I'd rather come up with a clean solution in V11 than fiddling something now.